sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -79,11 +79,12 @@
   /** minimum LogLevel to be used if debug is enabled */
   private @NotNull SentryLevel diagnosticLevel = DEFAULT_DIAGNOSTIC_LEVEL;
 
-  /** Serializer interface to serialize/deserialize json events */
-  private @NotNull ISerializer serializer = NoOpSerializer.getInstance();
-
+  /** Envelope reader interface */
   private @NotNull IEnvelopeReader envelopeReader = new EnvelopeReader();
 
+  /** Serializer interface to serialize/deserialize json events */
+  private @NotNull ISerializer serializer = new GsonSerializer(logger, envelopeReader);
```

Comment:
The only problem I see is that even if users provide their own serializer implementation, they are not able anymore to exclude Gson dependency. Which is maybe fine, i think we only need to make it explicit that Gson is a hard dependency from now on.

Reply:
that's true, ideally, we'd get rid of Gson altogether, and for that, we'd be refactoring this too, right now its suboptimal even as it is, as have a dependency and it won't work as well.